### DISCUSSION TITLE
To make "Assign if variable is not null, otherwise skip" be shorter

### DISCUSSION THREAD

**Discussion by Flithor** [dotnet/csharplang#4147] [Language Ideas] [answered]
at present
```
if(variable != null)
   target = variable;
```
the current shortest is
```
target = variable ?? target;
```
but it still assign one time when variable is null.
======

Test:
```
//Code:
target = variable ?? target;
//IL:
//ldloc.0
//dup
//brtrue.s	9 (000A) stloc.1 
//pop
//ldloc.1
//stloc.1
```
```
//Code:
if (variable != null)
    target = variable;
//IL:
//ldloc.0
//brfalse.s	14 (0010) ldloc.1 
```
consider to make it be shorter and faster?

**Comment by YairHalberstadt** [comment, marked as answer]
I think this should be opened at dotnet/Roslyn as it didn't require any language changes.

**Comment by canton7** [comment]
Those IL snippets look incomplete, and possibly also from a Debug build? Also, the length of the IL doesn't really mean much: check the ASM, and benchmark the two cases to see which is faster. 

Note that the IL will necessarily be different in those two cases, as the null coalescing operator guarantees that variable` is read only once.

**Comment by canton7** [comment]
(Copied to https://github.com/dotnet/roslyn/discussions/49427)
